File: db4oj/core/src/com/db4o/inside/QEBitmap.java
/**
 * 
 */
package com.db4o.inside;

import com.db4o.QE;

class QEBitmap {
	public static QEBitmap forQE(QE qe) {
    	boolean[] bitmap = new boolean[4];
    	qe.indexBitMap(bitmap);
    	return new QEBitmap(bitmap);
    }
	
	private QEBitmap(boolean[] bitmap) {
		_bitmap = bitmap;
	}
	
	private boolean[] _bitmap;

	public boolean takeGreater() {
		return _bitmap[QE.GREATER];
	}
	
	public boolean takeEqual() {
		return _bitmap[QE.EQUAL];
	}

	public boolean takeSmaller() {
		return _bitmap[QE.SMALLER];
	}
}

File: db4on/core/src/core/com/db4o/inside/fieldindex/QEBitmap.cs
namespace com.db4o.inside.fieldindex
{
	internal class QEBitmap
	{
		public static com.db4o.inside.fieldindex.QEBitmap ForQE(com.db4o.QE qe)
		{
			bool[] bitmap = new bool[4];
			qe.IndexBitMap(bitmap);
			return new com.db4o.inside.fieldindex.QEBitmap(bitmap);
		}

		private QEBitmap(bool[] bitmap)
		{
			_bitmap = bitmap;
		}

		private bool[] _bitmap;

		public virtual bool TakeGreater()
		{
			return _bitmap[com.db4o.QE.GREATER];
		}

		public virtual bool TakeEqual()
		{
			return _bitmap[com.db4o.QE.EQUAL];
		}

		public virtual bool TakeSmaller()
		{
			return _bitmap[com.db4o.QE.SMALLER];
		}
	}
}
